src/library/export/dlgtrackexport.cpp

```diff
@@ -0,0 +1,131 @@
+#include "library/export/dlgtrackexport.h"
+
+#include <QFileDialog>
+#include <QFileInfo>
+#include <QDesktopServices>
+#include <QMessageBox>
+
+#include "util/assert.h"
+
+DlgTrackExport::DlgTrackExport(QWidget *parent,
+                               ConfigObject<ConfigValue>* pConfig,
+                               QList<TrackPointer> tracks)
+        : QDialog(parent),
+          Ui::DlgTrackExport(),
+          m_pConfig(pConfig),
+          m_tracks(tracks) {
+    setupUi(this);
+    connect(cancelButton, SIGNAL(clicked()), this, SLOT(cancelButtonClicked()));
+    exportProgress->setMinimum(0);
+    exportProgress->setMaximum(tracks.size());
+    exportProgress->setValue(0);
+    statusLabel->setText("");
+    setModal(true);
+}
+
+bool DlgTrackExport::selectDestinationDirectory() {
+    QString lastExportDirectory = m_pConfig->getValueString(
+            ConfigKey("[Library]", "LastTrackCopyDirectory"),
+            QDesktopServices::storageLocation(QDesktopServices::MusicLocation));
+
+    QString destDir = QFileDialog::getExistingDirectory(
+            NULL, tr("Export Track Files To"), lastExportDirectory);
+    if (destDir.isNull() || destDir.isEmpty()) {
```

Comment:
btw, `destDir.isEmpty()` is confusing.
Do we talk about an empty dir (without any file in it) or an empty "destDir" string?
shouldn't destDir be a QDir instead of a QString?